commit 4520bcb2ba452863d744337db275c326835fca3f
Author: Arnd Bergmann <arnd@arndb.de>
Date:   Fri Aug 26 17:42:00 2016 +0200

    bug: Avoid Kconfig warning for BUG_ON_DATA_CORRUPTION
    
    The CONFIG_DEBUG_LIST option is normally meant for kernel developers
    rather than production machines and is guarded by CONFIG_DEBUG_KERNEL.
    
    In contrast, the newly added CONFIG_BUG_ON_DATA_CORRUPTION is meant
    for security hardening and may be used on systems that intentionally
    do not enable CONFIG_DEBUG_KERNEL. In this configuration, we get
    a warning from Kconfig about the mismatched dependencies:
    
    warning: (BUG_ON_DATA_CORRUPTION) selects DEBUG_LIST which has unmet direct dependencies (DEBUG_KERNEL)
    
    This annotates the DEBUG_LIST option to be selectable by
    BUG_ON_DATA_CORRUPTION when DEBUG_KERNEL is disabled.
    
    Signed-off-by: Arnd Bergmann <arnd@arndb.de>
    Fixes: 40cd725cfc7f ("bug: Provide toggle for BUG on data corruption")
    Acked-by: Rik van Riel <riel@redhat.com>
    Signed-off-by: Paul E. McKenney <paulmck@linux.vnet.ibm.com>

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index afa30fd52583..0c6366b77062 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1214,7 +1214,7 @@ config DEBUG_BUGVERBOSE
 
 config DEBUG_LIST
 	bool "Debug linked list manipulation"
-	depends on DEBUG_KERNEL
+	depends on DEBUG_KERNEL || BUG_ON_DATA_CORRUPTION
 	help
 	  Enable this to turn on extended checks in the linked-list
 	  walking routines.